tests/test_api.py

```diff
@@ -486,3 +485,157 @@ async def test_submit_job_fail_script_data26_non_standard(self):
+
+        # Error with desktop version lower than the minimum if the app is validating desktop version
+        resp = await self.client.request(
+            "POST", "/submit-job", json={"tx": tx_hex}, headers=header_desktop_lower
+        )
+        data = await resp.json()
+        if self.version_check:
+            self.assertEqual(400, resp.status)
+            self.assertEqual({"error": VERSION_CHECK_ERROR_MESSAGE}, data)
+        else:
+            self.assertEqual(200, resp.status)
+
+        header_mobile_lower = {"User-Agent": "Hathor Wallet Mobile / 0.18.0"}
+        header_mobile_equal = {"User-Agent": "Hathor Wallet Mobile / 1.18.3"}
+        header_mobile_higher = {"User-Agent": "Hathor Wallet Mobile / 20.1.0"}
+        header_mobile_custom = {"User-Agent": "HathorMobile/1"}
+        header_mobile_custom_wrong = {"User-Agent": "HathorMobile/2"}
+
+        # Success with mobile version higher than the minimum
+        resp = await self.client.request(
+            "POST", "/submit-job", json={"tx": tx_hex}, headers=header_mobile_higher
+        )
+        await resp.json()
+        self.assertEqual(200, resp.status)
+
+        # Success with mobile version equal to the minimum
+        resp = await self.client.request(
+            "POST", "/submit-job", json={"tx": tx_hex}, headers=header_mobile_equal
+        )
+        await resp.json()
+        self.assertEqual(200, resp.status)
+
+        # Error with mobile version lower than the minimum if the app is validating mobile version
+        resp = await self.client.request(
+            "POST", "/submit-job", json={"tx": tx_hex}, headers=header_mobile_lower
+        )
+        data = await resp.json()
+        if self.version_check:
+            self.assertEqual(400, resp.status)
+            self.assertEqual({"error": VERSION_CHECK_ERROR_MESSAGE}, data)
+        else:
+            self.assertEqual(200, resp.status)
+
+        # Error with mobile version before v0.18.0, if the app is validating mobile version
+        resp = await self.client.request(
+            "POST", "/submit-job", json={"tx": tx_hex}, headers=header_mobile_custom
+        )
+        data = await resp.json()
+        if self.version_check:
+            self.assertEqual(400, resp.status)
+            self.assertEqual({"error": VERSION_CHECK_ERROR_MESSAGE}, data)
+        else:
+            self.assertEqual(200, resp.status)
+
+        # Success if the custom is different than expected
+        resp = await self.client.request(
+            "POST",
+            "/submit-job",
+            json={"tx": tx_hex},
+            headers=header_mobile_custom_wrong,
+        )
+        await resp.json()
+        self.assertEqual(200, resp.status)
+
+        header_headless_lower = {"User-Agent": "Hathor Wallet Headless / 0.14.87"}
+        header_headless_equal = {"User-Agent": "Hathor Wallet Headless / 0.14.88"}
+        header_headless_higher = {"User-Agent": "Hathor Wallet Headless / 0.14.89"}
```

Comment:
Have you added tests for invalid versions? For example: Hathor Wallet Headless / 0.xx.88 or Hathor Wallet Headless / 0.14.88beta?

Reply:
Done https://github.com/HathorNetwork/tx-mining-service/pull/81/commits/2c6c6d6a74a8af506c02d9839f0ecacb4a463c2f

Reply:
Thanks!